tensorflow/python/distribute/distribute_lib.py

```diff
@@ -633,17 +638,20 @@ class InputOptions(
       strategy.experimental_distribute_dataset(
           dataset,
           tf.distribute.InputOptions(
-              experimental_prefetch_to_device=False)))
+              experimental_prefetch_to_device=False,
+              replication_mode=InputReplicationMode.PER_WORKER)))
   ```
```

Comment:
an example here for PER_REPLICA mode would be useful?

Reply:
ping

Reply:
Would you accept it if I add a DALI example here? Technically that is the only dataset derivation that supports PER_REPLICA as of now.

Reply:
I don't think you need DALI for an example. You can just have any simple dataset_fn and use PER_REPLICA mode. Same for unit tests - this code should work without DALI.